flood_adapt/object_model/hazard/event/event_set.py

```diff
@@ -0,0 +1,91 @@
+import os
+from pathlib import Path
+from typing import Any, List
+
+from pydantic import Field, model_validator
+from typing_extensions import Annotated
+
+from flood_adapt.object_model.hazard.event.synthetic import SyntheticEventModel
+from flood_adapt.object_model.hazard.event.template_event import (
+    EventModel,
+)
+from flood_adapt.object_model.hazard.interface.events import IEvent, Mode
+from flood_adapt.object_model.interface.database_user import DatabaseUser
+from flood_adapt.object_model.interface.object_model import IObject, IObjectModel
+
+
+class EventSetModel(IObjectModel):
+    """BaseModel describing the expected variables and data types for parameters of Synthetic that extend the parent class Event."""
+
+    mode: Mode = Mode.risk
+
+    sub_events: List[EventModel]
+    frequency: List[Annotated[float, Field(strict=True, ge=0, le=1)]]
+
+    @model_validator(mode="before")
+    def load_sub_events(self):
+        """Load the sub events from the dictionary."""
+        from flood_adapt.object_model.hazard.event.event_factory import EventFactory
```

Comment:
Due to circular imports causing problems that I didnt feel like solving in the moment. 
It was problematic that an EventSet was an Event as well, even though they do quite different things. 
Single Event: 1 sfincs run + 1 fiat run
EventSet: any nr of sfincs runs + 1 fiat run

Because there was no distinction between these two approaches, we have everywhere in the code if statements like this: 
```python
if event.attrs.mode == risk: 
  ...
else:
  ...
```
or 
```python
isinstance(scenario.event, EventSet):
  ...
```

This shouldnt happen imo.

Making the EventSets a separate class similar to the scenario / benefits class, that contains a set of the new Events, will make it much better. But that is for a future PR